customer/src/main/java/no/unit/nva/customer/get/ListAllChannelClaimsHandler.java

```
@@ -67,12 +66,29 @@ private static String decode(String cristinId) {
         return URLDecoder.decode(cristinId, StandardCharsets.UTF_8);
     }
 
-    @Override
-    protected Integer getSuccessStatusCode(Void unused, ChannelClaimsListResponse o) {
-        return HTTP_OK;
+    private static Optional<String> getType(RequestInfo requestInfo) {
+        return requestInfo.getQueryParameterOpt(QUERY_PARAM_TYPE);
     }
 
     private static void userIsAuthorized(RequestInfo requestInfo) throws UnauthorizedException {
         requestInfo.getCurrentCustomer();
     }
+
+    private Collection<ChannelClaimWithClaimer> listChannelClaims(RequestInfo requestInfo) {
+        var claims = getInstitutionCristinId(requestInfo)
+                         .map(customerService::getChannelClaimsForCustomer)
+                         .orElse(customerService.getChannelClaims());
+
+        return getType(requestInfo)
+                   .map(type -> filterByType(claims, type))
+                   .orElse(claims);
+    }
+
+    private Collection<ChannelClaimWithClaimer> filterByType(Collection<ChannelClaimWithClaimer> claims, String type) {
+        return claims.stream().filter(claim -> isType(claim, type)).toList();
+    }
+
+    private boolean isType(ChannelClaimWithClaimer claim, String type) {
+        return claim.channelClaim().channel().toString().contains(type);
```

Can claim::channelClaim or subsequent call to channel() and toString() cause NPEs?

---

[Line 33 in the Validator](https://github.com/BIBSYSDEV/nva-identity-service/blob/f11aecc78c185c3d1508680afd8aff946a3ba6dc/customer-commons/src/main/java/no/unit/nva/customer/validator/ChannelClaimValidator.java#L33) should throw an exception if no channel is given. But an extra check here doesn't hurt.